python/pyspark/sql/tests/test_functions.py

```diff
@@ -709,6 +709,52 @@ def test_overlay(self):
             message_parameters={"arg_name": "len", "arg_type": "float"},
         )
 
+    def test_percentile(self):
+        actual = list(
+            chain.from_iterable(
+                [
+                    re.findall("(percentile\\(.*\\))", str(x))
+                    for x in [
+                        F.percentile(F.col("foo"), F.lit(0.5)),
+                        F.percentile(F.col("bar"), 0.25, 2),
+                        F.percentile(F.col("bar"), [0.25, 0.5, 0.75]),
+                        F.percentile(F.col("foo"), (0.05, 0.95), 100),
+                        F.percentile("foo", 0.5),
+                        F.percentile("bar", [0.1, 0.9], F.lit(10)),
+                    ]
+                ]
+            )
+        )
+
+        expected = [
+            "percentile(foo, 0.5, 1)",
+            "percentile(bar, 0.25, 2)",
+            "percentile(bar, array(0.25, 0.5, 0.75), 1)",
+            "percentile(foo, array(0.05, 0.95), 100)",
+            "percentile(foo, 0.5, 1)",
+            "percentile(bar, array(0.1, 0.9), 10)",
+        ]
+
+        self.assertListEqual(actual, expected)
+
+    def test_median(self):
```

Comment:
why test `median`? it already exists

https://github.com/apache/spark/blob/2057eb7e203c9fde3f4fa13d5f04225cf6e49a87/python/pyspark/sql/functions.py#L701

Reply:
Because median is a variant of percentile and Spark missing some test cases for it, so this PR also supplement some test cases.